### ISSUE TITLE
Waypoints with invalid pose

### ISSUE THREAD

**Issue by jim-v** [UbiquityRobotics/Robot_Commander#3]
If a waypoint is set when the robot location is unknown, then no-error is reported, but when asked to go to that waypoint, RobotCommander sends the following goal:

header: 
  seq: 16
  stamp: 
    secs: 0
    nsecs:         0
  frame_id: ''
goal_id: 
  stamp: 
    secs: 0
    nsecs:         0
  id: goal_0.06402862432775747_1494305466006
goal: 
  target_pose: 
    header: 
      seq: 0
      stamp: 
        secs: 1494305466
        nsecs: 619306087
      frame_id: /map
    pose: 
      position: 
        x: 0.0
        y: 0.0
        z: 0.0
      orientation: 
        x: 0.0
        y: 0.0
        z: 0.0
        w: 0.0

**Comment by jrlandau** [comment]
So what is the desired Robot Commander behavior?  Is it that if x=y=z=0
then show an error message and do not save the waypoint?

**Comment by jim-v** [comment]
There should be an exception thrown if the current pose can't be determined.  You should be able to catch that and give the user an error message that the waypoint can't be saved because the robot's position is not known.
x=0=y=0 is a valid position, but the quaternion with all elements set to 0 is not valid.

**Comment by rohbotics** [comment]
I think If there is no pose data available, it shoudn't default to 0s, it should error when setting the waypoint 

Regardless, It shouldn't send an invalid goal
The orientation is a quaternion and 
```
x: 0.0
y: 0.0
z: 0.0
w: 0.0
``` 
is not a valid quaternion
The valid quaterion for zero is 
```
x: 0.0
y: 0.0
z: 0.0
w: 1.0
```

**Comment by jrlandau** [comment]
For testing purposes, how to provoke this situation?  On magni George, or
on Stage?

**Comment by davecrawley** [comment]
Agree, invalid pose should draw human attention to the error. (Poka-yoke)

**Comment by jim-v** [comment]
On a robot, don't run localization, or cover the camera, or turn the lights off, then set a waypoint.

**Comment by jrlandau** [comment]
If an invalid pose is received, the error is caught and a response is given to the user informing her.  If a pose is requested and it is never received, the failure is silent.
